fedora_bot.py

```diff
@@ -101,24 +102,37 @@ def slack_notify(message: str):
     assert response.body == "ok"
 
 
-def check_pull_request_flags(component, pr_id, num_tests):
+def fedora_distgit_http_client() -> requests.Session:
+    """
+    Returns a configured http client for Fedora's dist-git. We need a custom
+    client in order to implement retries. These are needed because
+    src.fedoraproject.org proved to be slightly unreliable and without
+    retries, some fedora-bot runs failed.
+    """
+    retry_adapter = Retry(
+        total=3,
+        backoff_factor=0.1,
+        # POST isn't generally idempotent but we only use it to merge PRs
+        # which actually idempotent is.
+        allowed_methods=["POST"] + list(Retry.DEFAULT_ALLOWED_METHODS),
+        # Enable retrying on retryable http codes.
+        status_forcelist=[500, 502, 502, 503, 504],
```

Comment:
Duplicate 502 -- this looks like you meant 501?

Reply:
Nah, 501 is Not Implemented, this is not retryable. I want:

- 500 Internal Server Error
- 502 Bad Gateway
- 503 Service Unavailable
- 504 Gateway timeout